### PR TITLE
Use publicUrl to match files from our storage service

### PR DESCRIPTION
**Pull request by Jenselme**
I directly use publicUrl so it can only match one domain. I don't know if you have the case where files can come from both public.admin.ch and public.bgdi.ch. If so, let me know, I will update the pull request.

### CODE REVIEW

**Comment by oterral** [comment]
yes we have this use case where files can come from both public.admin.ch and public.bgdi.ch

**Comment by Jenselme** [comment]
I see two possibility to cover this case:
1. Add an `publicAllowedUrlRegexp` option in configuration. In your case it would be `https?:\/\/public\..*(\.admin\.ch|\.bgdi\.ch)\/.*`. This is the closest to the current code.
2. Add a `publicAllowedUrls` option in configuration which would contain an array of allowed urls, in your case `['public.admin.ch', 'public.bgdi.ch']`.

I don't know which one is the best. @oterral Do you have an opinion?

**Comment by oterral** [comment]
@jenselme a regex is more powerful :)

**Comment by Jenselme** [comment]
A regexp it is then. PR updated.

**Comment by Jenselme** [comment]
I just updated this PR to take into account the new hard coded URLs from  src/components/UrlUtilsService.js

File: Makefile [hunk 1]
```diff
@@ -9,6 +9,9 @@ LAST_APACHE_BASE_PATH := $(shell if [ -f .build-artefacts/last-apache-base-path
 API_URL ?= //mf-chsdi3.dev.bgdi.ch
 LAST_API_URL := $(shell if [ -f .build-artefacts/last-api-url ]; then cat .build-artefacts/last-api-url 2> /dev/null; else echo '-none-'; fi)
 PUBLIC_URL ?= //public.dev.bgdi.ch
+PUBLIC_ALLOWED_URL_REGEXP ?= https?:\/\/public\..*(\.admin\.ch|\.bgdi\.ch)\/.*
+PUBLIC_URL_REGEXP ?= ^https?:\/\/public\..*(bgdi|geo\.admin)\.ch.*
```
Comment (oterral) [inline Makefile]: `PUBLIC_ALLOWED_URL_REGEXP`  and `PUBLIC_URL_REGEXP` do the same thing . Can you remove `PUBLIC_ALLOWED_URL_REGEXP` pls ?

File: src/components/map/MapService.js [hunk 2]
```diff
@@ -1263,8 +1263,8 @@ goog.require('ga_urlutils_service');
           if (id instanceof ol.layer.Layer) {
             id = olLayerOrId.id;
           }
-          var regex = /https?:\/\/public\..*(\.admin\.ch|\.bgdi\.ch)\/.*/;
-          return this.isKmlLayer(olLayerOrId) && regex.test(id);
+          return this.isKmlLayer(olLayerOrId) &&
+                  gaGlobalOptions.publicAllowedUrlRegexp.test(id);
```
Comment (oterral) [inline src/components/map/MapService.js]: Can you add a function `isPublicValid` in `gaUrlUtils` pls then use it here.

File: src/index.mako.html [hunk 3]
```diff
@@ -686,6 +686,8 @@
           mapUrl : location.origin + '${apache_base_path}',
           apiUrl : location.protocol + '${api_url}',
           publicUrl : location.protocol + '${public_url}',
+          publicAllowedUrlRegexp: new RegExp('${public_allowed_url_regexp}'),
```
Comment (oterral) [inline src/index.mako.html]: To remove

File: test/specs/Loader.spec.js [hunk 4]
```diff
@@ -33,8 +33,10 @@ beforeEach(function() {
       ],
       defaultTopicId: 'sometopic',
       translationFallbackCode: 'somelang',
-      defaultExtent: [420000, 30000,900000, 350000],
```
Comment (oterral) [inline test/specs/Loader.spec.js]: Why removing `defaultExtent`?

File: test/specs/Loader.spec.js [hunk 5]
```diff
@@ -33,8 +33,10 @@ beforeEach(function() {
...
-      defaultResolution: 500.0
+      defaultResolution: 500.0,
+      publicAllowedUrlRegexp: /https?:\/\/public\..*(\.admin\.ch|\.bgdi\.ch)\/.*/,
```
Comment (oterral) [inline test/specs/Loader.spec.js]: to remove

File: Makefile (hunk 1 above)
Comment (Jenselme) [inline Makefile]: Maybe that both regexp should do the same thing but they currently don't. The first will match `https://public.bgdi.ch` but not the latter. Maybe you can give me the regexp that matches all your cases so I can easily use your test suite to test this PR.

File: test/specs/Loader.spec.js (hunk 4 above)
Comment (Jenselme) [inline test/specs/Loader.spec.js]: It's a mistake, I had `defaultExtent` twice in my branch. I will cancel the deletion.

**Comment by Jenselme** [comment]
@oterral  I will wait for #2344 before rebasing and taking into account your suggestions.

**Comment by Jenselme** [comment]
While removing `PUBLIC_ALLOWED_URL_REGEXP ?= https?:\/\/public\..*(\.admin\.ch|\.bgdi\.ch)\/.*` I compared `PUBLIC_URL_REGEXP ?= ^https?:\/\/public\..*(bgdi|geo\.admin)\.ch.*` and `ADMIN_URL_REGEXP ?= ^(ftp|http|https):\/\/(.*(\.bgdi|\.geo\.admin)\.ch)` and they look pretty similar to me.

Can/should they be merged in one regexp?

**Comment by Jenselme** [comment]
@oterral PR updated with your remarks. `ga_map_service gaMapUtils tests isStoredKmlLayer` fail because the two regexps are not exactly the same. Since you know better than me the URLs that should match, can you write it?

**Comment by oterral** [comment]
To summarize, we need to detect if a external file comes from our domain (*.admin.ch and *.bgdi.ch) and if it comes from a third party domain ( all hostnames + public.*.admin.ch + public.*.bgdi.ch)

**Comment by Jenselme** [comment]
In `public..admin.ch` and `public..bgdi.ch` is the double dot `..` intentional or is a star missing?

**Comment by oterral** [comment]
A star is missing :)

**Comment by oterral** [comment]
The test failed [here](https://github.com/geoadmin/mf-geoadmin3/pull/2494/files#diff-66ac86f99915855fde57c0a20d8f6d6bL1267) because id is like `KML||https://public.geo.admin.ch` and not `https://public.geo.admin.ch`

**Comment by oterral** [comment]
I will fix it in another PR

**Comment by Jenselme** [comment]
Just updated the regexp and fix `ga_map_service gaMapUtils tests isStoredKmlLayer`

File: src/index.mako.html [hunk 6]
```diff
@@ -686,6 +686,7 @@
           mapUrl : location.origin + '${apache_base_path}',
           apiUrl : location.protocol + '${api_url}',
           publicUrl : location.protocol + '${public_url}',
+          adminUrlRegexp: new RegExp('${admin_url_regexp}'),
```
Comment (oterral) [inline src/index.mako.html]: publicUrlRegExp is missing

Comment (Jenselme) [inline src/index.mako.html]: Corrected.

**Comment by cedricmoullet** [comment]
@Jenselme could you please resolve the conflicts ?

**Comment by Jenselme** [comment]
Conflicts resolved!

File: src/index.mako.html (hunk 6 above)
Comment (oterral) [inline src/index.mako.html]: Forget to push ?

Comment (Jenselme) [inline src/index.mako.html]: So it seems. Just pushed the modifications.

**Comment by oterral** [comment]
@Jenselme it seems you p have pushed an old version of the PR because the isPublicValid function has disappeared.

**Comment by Jenselme** [comment]
This time it should be good.

File: test/specs/Loader.spec.js (hunk 4 above)
Comment (oterral) [inline test/specs/Loader.spec.js]: Can you re-add defaultExtent pls

**Comment by oterral** [comment]
@Jenselme we are close to the end still 2 things:

 - [ ] another rebase because of https://github.com/geoadmin/mf-geoadmin3/pull/2541
- [ ] add defaultExtent in loader.spec.js

**Comment by Jenselme** [comment]
>  add defaultExtent in loader.spec.js 

If I look ad loader.spec.js, line 24 I see `defaultExtent`. So I guess that this key was present two times before. Should I readd it anyway?

PR rebased.

**Comment by oterral** [comment]
Ah yes you're right so it's good .

**Comment by oterral** [comment]
Thx for your patience.

**Pull request merged**